### PR TITLE
Server configuration

### PR DESCRIPTION
**Pull request by stevenjack**
### Problem

Currently the configuration is via cli parameters, would be good to follow the same structure as https://github.com/neo-project/neo/blob/master/neo/protocol.json and https://github.com/CityOfZion/neo-python/blob/master/protocol.privnet.json. Also would be good go set the version and time of build when building the binary.

### Solution

* Add config struct with a mix of the official `neo` config and `neo-python` config.
* Add convenience methods on config for checking for available seeds and valid net mode.
* Use config in server. 

### Notes

- Looking to add env var overrides so you can override values easily.
- Use [semver](https://semver.org/) to bump `VERSION`, and remember to run the following after
merging your PR:

```
make push-tag
```

### CODE REVIEW

File: pkg/wallet/transfer_output.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package wallet
+
+import (
+	"math/big"
+
+	"github.com/CityOfZion/neo-go/pkg/util"
+)
+
+// TransferOutput respresents the output of a transaction.
+type TransferOutput struct {
```
Comment (anthdm) [inline pkg/wallet/transfer_output.go]: The `transaction` package inside `core` has an `Output` type that represents the output of a transaction.

**Comment by anthdm** [comment]
@stevenjack Just merged the p2p refactor with the master branch, would be nice if this gets updated according to the changes in master. 👍

**Comment by anthdm** [comment]
This looks very good, will take an in-depth look this evening. Are you planning to make the config file yaml based? Or just leave it as JSON?

**Comment by stevenjack** [comment]
@anthdm Yep will move it over to `yaml` 👍

**Comment by anthdm** [comment]
Nicely done 👍

**Pull request merged**